pkg/manager/member/utils.go

```diff
@@ -55,13 +55,13 @@ func annotationsMountVolume() (corev1.VolumeMount, corev1.Volume) {
 
 // statefulSetIsUpgrading confirms whether the statefulSet is upgrading phase
 func statefulSetIsUpgrading(set *apps.StatefulSet) bool {
-	if set.Status.ObservedGeneration == nil {
+	if &(set.Status.ObservedGeneration) == nil {
```

Comment:
This looks weird, why change this?

Reply:
The `ObservedGeneration` property defined in `StatefulSetStatus` in `v1beta1` package was a pointer type for`int64` while it was `int64` type in `v1` package.
```golang
package v1beta1

type StatefulSetStatus struct {
	// observedGeneration is the most recent generation observed for this StatefulSet. It corresponds to the
	// StatefulSet's generation, which is updated on mutation by the API Server.
	// +optional
	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
        ....
}
```
```golang
package v1

// StatefulSetStatus represents the current state of a StatefulSet.
type StatefulSetStatus struct {
	// observedGeneration is the most recent generation observed for this StatefulSet. It corresponds to the
	// StatefulSet's generation, which is updated on mutation by the API Server.
	// +optional
	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
        ....
}
```

Reply:
this statement can be removed because it's a noop operation if ObservedGeneration is an int64 type.

Reply:
@Yisaer this should be cleaned too

Reply:
The &() will never be nil.

Reply:
fix